ansible_collections/arista/avd/docs/semantic-versioning.md

```diff
@@ -0,0 +1,216 @@
+
+### Role: eos_config_deploy_cvp
+
+Inputs:
+
+- All role input variables follow SemVer.
+  - Any breaking changes will at a minimum be communicated with a deprecation notice for 6 months.
+  - Keys will then be documented as removed in the next major release.
+
+Outputs:
+
+- Structured configuration output may change during minor releases to accommodate changes in the ansible-cvp collection.
+- The resulting CloudVision configuration will be maintained, unless in rare cases, when addressing a bug.
+  - Breaking changes will be communicated in the release notes.
+
+### Role: cv_deploy
+
+Inputs:
+
+- All role input variables follow SemVer.
+  - Any breaking changes will at a minimum be communicated with a deprecation notice for 6 months.
+  - Keys will then be documented as removed in the next major release.
+
+Outputs:
+
+- The resulting CloudVision configuration will be maintained, unless in rare cases, when addressing a bug.
+  - Breaking changes will be communicated in the release notes.
+
+### Role: eos_config_deploy_eapi
+
+Inputs:
+
+- All role input variables follow SemVer.
+  - Any breaking changes will at a minimum be communicated with a deprecation notice for 6 months.
+  - Keys will then be documented as removed in the next major release.
+
+Outputs:
+
+- All role outputs follow SemVer.
+
+### Role: eos_validate_state
+
+Inputs:
+
+- All role input variables follow SemVer.
+  - Any breaking changes will at a minimum be communicated with a deprecation notice for 6 months.
+  - Keys will then be documented as removed in the next major release.
+
+Outputs:
+
+- The generated reports and other artifacts may change between minor releases.
+  - Breaking changes to YAML/JSON/CSV outputs will be communicated in the release notes.
+
+### Role: dhcp_provisioner
+
+Inputs:
+
+- All role input variables follow SemVer.
+  - Any breaking changes will at a minimum be communicated with a deprecation notice for 6 months.
+  - Keys will then be documented as removed in the next major release.
+
+Outputs:
+
+- Structured configuration output may change during minor releases to accommodate changes in the ansible-cvp collection.
+- The resulting DHCP configuration will be maintained, unless in rare cases, when addressing a bug.
+  - Breaking changes will be communicated in the release notes.
+
+### Role: build_output_folders
+
+Inputs:
+
+- All role input variables follow SemVer.
+  - Any breaking changes will at a minimum be communicated with a deprecation notice for 6 months.
+  - Keys will then be documented as removed in the next major release.
+
+Outputs:
+
+- All role ouptus follow SemVer.
+
+### Ansible Plugins
+
+- No Ansible modules, action plugins, filter plugins, test plugins, or vars plugins are covered by SemVer.
+  - Functionality may change in minor releases to accommodate the project's needs.
+  - Breaking changes will be communicated in the release notes.
+
+## PyAVD
```

Comment:
Maybe a table for pyavd and a table for ansible avd with a row per role/function would be simpler?

Reply:
Great suggestion, I converted it to tables!